src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AuditLogImpl.java

```diff
@@ -85,23 +86,11 @@ public AuditLogImpl(final Settings settings,
 			sm.checkPermission(new SpecialPermission());
 		}
 
-		AccessController.doPrivileged(new PrivilegedAction<Object>() {
-			@Override
-			public Object run() {
-				Runtime.getRuntime().addShutdownHook(new Thread() {
-
-					@Override
-					public void run() {
-						try {
-							close();
-						} catch (final IOException e) {
-							log.warn("Exception while shutting down message router", e);
-						}
-					}
-				});
-				log.debug("Shutdown Hook registered");
-				return null;
-			}
+		shutdownHook = new Thread(() -> messageRouter.close());
```

Comment:
Shouldn't this be ```this.close()``` instead of ```messageRouter.close()``` ?

Reply:
We can't call `this.close()` from the shutdown hook as it calls `removeShutdownHook()` while shutdown is in process.